### PR TITLE
Make the default value of credentialsManagerEnabled false

### PR DESCRIPTION
**Pull request by weiqingy**
## What changes were proposed in this pull request?
This PR is to make the default value of credentialsManagerEnabled false.

cc: @mridulm @jerryshao @bikassaha

### CODE REVIEW

**Comment by weiqingy** [comment]
Hi, @mridulm 

In the [PR#120](https://github.com/hortonworks-spark/shc/pull/120)  ("Modify the implementation of Supporting multiple Hbase secure clusters"), it set [spark.hbase.connector.security.credentials.enabled](https://github.com/hortonworks-spark/shc/blob/master/core/src/main/scala/org/apache/spark/sql/execution/datasources/hbase/SHCCredentialsManager.scala#L59) to true by default. But with [Jerry's PR ](https://github.com/hortonworks-spark/shc/pull/122) in, if SHCCredentialManager is enabled, users are required to set key and keytab no matter if applications are long running or just common ones ([here](https://github.com/hortonworks-spark/shc/blob/master/core/src/main/scala/org/apache/spark/sql/execution/datasources/hbase/SHCCredentialsManager.scala#L77)). 

This PR make the default value of `credentialsManagerEnabled` false, otherwise, if users want to use Spark internal HBaseCredentialManager, they'll be asked to do extra work: set key and keytab or set `spark.hbase.connector.security.credentials.enabled ` = false. What do you think of this? Thanks.

File: README.md [hunk 1]
```diff
@@ -159,6 +160,22 @@ or
     
 Include the hbase-site.xml under SPARK_CONF_DIR (/etc/spark/conf) on the host where the spark job is submitted from, by creating a symbolic link towards your main hbase-site.xml (in order to be synchronous with your platform updates).
 
+### Using SHCCredentialsManager
+
+Spark only supports use cases which access a single secure HBase cluster. If your applications need to access multiple secure HBase clusters, users need to use SHCCredentialsManager instead.
+SHCCredentialsManager supports a single secure HBase cluster as well as multiple secure HBase clusters. It is disabled by default, but users can set _spark.hbase.connector.security.credentials.enabled_ to true to enable it.
+Also, users need to config principal and keytab as below before running their applications.
+
```
Comment (jerryshao) [inline README.md]: I would suggest to move below 3 lines to here:

```
spark.hbase.connector.security.credentials.enabled true
spark.hbase.connector.security.credentials  ambari-qa-c1@EXAMPLE.COM
spark.hbase.connector.security.keytab  /etc/security/keytabs/smokeuser.headless.keytab
```

I remembered we took `spark.hbase.connector.security.credentials` a precedence over `spark.yarn.principal` from the code, would be better to also reflect this in the docs.

**Comment by jerryshao** [comment]
LGTM.

**Comment by mridulm** [comment]
Other than @jerryshao's suggestions, LGTM

**Comment by weiqingy** [comment]
Thanks for the review!

File: README.md (hunk 1 above)
Comment (weiqingy) [inline README.md]: Done.

File: README.md [hunk 2]
```diff
@@ -135,8 +135,9 @@ Users can include the package as the dependency in your SBT file as well. The fo
 
 For running in a Kerberos enabled cluster, the user has to include HBase related jars into the classpath as the HBase token
 retrieval and renewal is done by Spark, and is independent of the connector. In other words, the user needs to initiate the
-environment in the normal way, either through kinit or by providing principal/keytab.  The following examples show how to run
-in a secure cluster with both yarn-client and yarn-cluster mode. Note that SPARK_CLASSPATH has to be set for both modes.
+environment in the normal way, either through kinit or by providing principal/keytab. The following examples show how to run
+in a secure cluster with both yarn-client and yarn-cluster mode. Note that if your Spark does not contain [SPARK-20059](https://github.com/apache/spark/pull/17388),
+which is in Apache Spark 2.1.1+, you need to set SPARK_CLASSPATH for both modes.
 
     export SPARK_CLASSPATH=/usr/hdp/current/hbase-client/lib/hbase-common.jar:/usr/hdp/current/hbase-client/lib/hbase-client.jar:/usr/hdp/current/hbase-client/lib/hbase-server.jar:/usr/hdp/current/hbase-client/lib/hbase-protocol.jar:/usr/hdp/current/hbase-client/lib/guava-12.0.1.jar
```
Comment (bikassaha) [inline README.md]: Lets move this to a wiki page and reference that wiki here in a link.

Comment (weiqingy) [inline README.md]: Sure.

**Comment by weiqingy** [comment]
Merged. Will make a wiki page for ` SPARK_CLASSPATH`  part.

**Pull request merged**